### PR TITLE
BS-14519, fix entityGetter return type when method returns a list

### PR DESCRIPTION
**Pull request by laurentleseigneur**

### CODE REVIEW

File: services/bonita-business-data/bonita-business-data-client-resources/src/test/java/org/bonitasoft/engine/bdm/dao/client/resources/proxy/LazyLoaderTest.java [hunk 1]
```diff
@@ -72,17 +74,39 @@ public void should_load_list_of_objects_through_command_api() throws Exception {
...
         Object loadedChild = lazyLoader.load(getChildren, persistenceId);
-        
+
         assertThat(loadedChild).isInstanceOf(List.class);
         assertThat((List<Child>) loadedChild).containsOnly(luce, julien);
     }
-    
+
+    @Test
+    public void should_getParameters_returnType_when_return_list_is_true() throws Exception {
```
Comment (ermedeiros) [inline services/bonita-business-data/bonita-business-data-client-resources/src/test/java/org/bonitasoft/engine/bdm/dao/client/resources/proxy/LazyLoaderTest.java]: Maybe the test name can be improved.

File: services/bonita-business-data/bonita-business-data-client-resources/src/test/java/org/bonitasoft/engine/bdm/dao/client/resources/utils/EntityGetterTest.java [hunk 2]
```diff
@@ -26,88 +26,121 @@
...
+        //given
         Method multipleObjectGetter = Employee.class.getMethod("getAddresses");
 
-        String classname = new EntityGetter(multipleObjectGetter).getReturnTypeClassName();
+        //when
+        String className = new EntityGetter(multipleObjectGetter).getReturnTypeClassName();
 
-        assertThat(classname).isEqualTo(List.class.getName());
+        //then
+        assertThat(className).isNotEqualTo(List.class.getName());
```
Comment (ermedeiros) [inline services/bonita-business-data/bonita-business-data-client-resources/src/test/java/org/bonitasoft/engine/bdm/dao/client/resources/utils/EntityGetterTest.java]: why not assertThat(classname).isEqualTo(Address.class.getName()) ?

File: services/bonita-business-data/bonita-business-data-client-resources/src/test/java/org/bonitasoft/engine/bdm/dao/client/resources/proxy/LazyLoaderTest.java (hunk 1 above)
Comment (laurentleseigneur) [inline services/bonita-business-data/bonita-business-data-client-resources/src/test/java/org/bonitasoft/engine/bdm/dao/client/resources/proxy/LazyLoaderTest.java]: should_getParameters_return_real_type_when_query_returns_list ?

File: services/bonita-business-data/bonita-business-data-client-resources/src/test/java/org/bonitasoft/engine/bdm/dao/client/resources/utils/EntityGetterTest.java (hunk 2 above)
Comment (laurentleseigneur) [inline services/bonita-business-data/bonita-business-data-client-resources/src/test/java/org/bonitasoft/engine/bdm/dao/client/resources/utils/EntityGetterTest.java]: may be this test can be merged with should_be_able_to_retrieve_target_entity_class_even_if_method_return_a_list

File: services/bonita-business-data/bonita-business-data-client-resources/src/test/java/org/bonitasoft/engine/bdm/dao/client/resources/proxy/LazyLoaderTest.java (hunk 1 above)
Comment (ermedeiros) [inline services/bonita-business-data/bonita-business-data-client-resources/src/test/java/org/bonitasoft/engine/bdm/dao/client/resources/proxy/LazyLoaderTest.java]: ok, for me.

File: services/bonita-business-data/bonita-business-data-client-resources/src/test/java/org/bonitasoft/engine/bdm/dao/client/resources/utils/EntityGetterTest.java (hunk 2 above)
Comment (ermedeiros) [inline services/bonita-business-data/bonita-business-data-client-resources/src/test/java/org/bonitasoft/engine/bdm/dao/client/resources/utils/EntityGetterTest.java]: Good idea!

**Comment by laurentleseigneur** [comment]
@ermedeiros also refactored client side EntityGetterTest

**Pull request merged**